testlapack/general.go

```diff
@@ -1038,3 +1038,85 @@ func unbalancedSparseGeneral(m, n, stride int, nonzeros int, rnd *rand.Rand) bla
 	}
 	return a
 }
+
+func columnOf(a blas64.General, j int) []float64 {
+	if j < 0 || a.Cols <= j {
+		panic("bad column index")
+	}
+	col := make([]float64, a.Rows)
+	for i := range col {
+		col[i] = a.Data[i*a.Stride+j]
+	}
+	return col
+}
+
+func isRightEigenvectorOf(a blas64.General, xRe, xIm []float64, lambda complex128, tol float64) bool {
+	if a.Rows != a.Cols {
+		panic("matrix not square")
+	}
+
+	n := a.Rows
+	axRe := make([]float64, n)
+	blas64.Gemv(blas.NoTrans, 1, a, blas64.Vector{1, xRe}, 0, blas64.Vector{1, axRe})
+	if imag(lambda) == 0 {
+		// Real eigenvalue and eigenvector.
```

Comment:
Great observation. I was so submerged in `Dtrevc3` that I didn't consider this situation because it doesn't arise with Dtrevc3

Real matrix cannot have a real eigenvector corresponding to complex eigenvalue. I added a check for this to `is{Left,Right}EigenvectorOf`.

Real eigenvalue and complex eigenvector is possible (for example, multiply the real eigenvector by i) although a real eigenvector always exists in such case and that's what Dtrevc3 computes. I modified the checking routines to cover this.

Thanks. Done.